### PR TITLE
Fixed #29550 -- Made it easier to override pickle.dumps usages protocol.

### PR DESCRIPTION
**Pull request by charettes**
https://code.djangoproject.com/ticket/29550

### CODE REVIEW

File: django/core/cache/backends/filebased.py [hunk 1]
```diff
@@ -13,13 +13,14 @@
 from django.core.files.move import file_move_safe
 
 
-def _write_content(f, expiry, value):
-    f.write(pickle.dumps(expiry, pickle.HIGHEST_PROTOCOL))
-    f.write(zlib.compress(pickle.dumps(value, pickle.HIGHEST_PROTOCOL)))
+def _write_content(f, expiry, value, protocol):
```
Comment (sir-sigurd) [inline django/core/cache/backends/filebased.py]: Might make sense to rework this function as method.

Comment (charettes) [reply django/core/cache/backends/filebased.py]: @sir-sigurd agreed!

Comment (charettes) [reply django/core/cache/backends/filebased.py]: @sir-sigurd, @claudep handled in 1a1ed064635fd43fb9018e5d30af9452b337cdc5.

**Comment by charettes** [comment]
Thanks for the merge @timgraham!

**Pull request merged**